### PR TITLE
Update metabat2 output channels and compression

### PR DESCRIPTION
**Pull request by jfy133**
- update so each output type has dedicated channel 
- replace bgzip with gzip as bgzip can only zip one file at a time, so not all output was compressed

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Comment by jfy133** [comment]
Realised `mv` command doesn't work as the 'failed' bins arne't always generated

File: modules/metabat2/metabat2/main.nf [hunk 1]
```diff
@@ -33,8 +36,19 @@ process METABAT2_METABAT2 {
 
     mv metabat2/${prefix} ${prefix}.tsv
     mv metabat2 bins
-    bgzip --threads $task.cpus ${prefix}.tsv
-    bgzip --threads $task.cpus bins/*.fa
+
+    gzip ${prefix}.tsv
+    gzip bins/*.fa
```
Comment (maxibor) [inline modules/metabat2/metabat2/main.nf:41]: Did some xargs diggging (the container version of xargs is old).
This should work
```suggestion
    find . -type f -name "bins/*.fa" | xargs -t -n 1 bgzip -@ 4
```

Comment (jfy133) [reply modules/metabat2/metabat2/main.nf]: Will try on my mag test

File: modules/metabat2/metabat2/main.nf [hunk 2]
```diff
@@ -33,8 +36,19 @@ process METABAT2_METABAT2 {
...
+
+    ## Some bash trickery to successfully execute `mv`, even when no contigs are
+    ## actually discarded.
+    ## Thanks to Stéphane Chazelas https://unix.stackexchange.com/a/522283/478356
+    (
+        shopt -s nullglob
+        shopt -u failglob
+        files=(bins/*{lowDepth,tooShort,unbinned}.fa.gz)
+        ((\${#files[@]} == 0)) || mv -- "\${files[@]}" .
+    )
```
Comment (maxibor) [inline modules/metabat2/metabat2/main.nf:51]: What about using `find` ?

Something like:

```bash
find ./bins/ -type f -name "*{lowDepth,tooShort,unbinned}.fa.gz" -exec mv {} . \;
```

Comment (jfy133) [reply modules/metabat2/metabat2/main.nf:51]: Didn't work when I tried it. When none of the files do not exist, it fails.

Comment (jfy133) [reply modules/metabat2/metabat2/main.nf:51]: Ok now it works... :confused:

Comment (jfy133) [reply modules/metabat2/metabat2/main.nf]: Fix: needed sqaure brackets not curly

**Review by maxibor — APPROVED** [review]

**Pull request merged**